### PR TITLE
Update LR step scheduler to use total step to work across epochs

### LINKED ISSUE #17544: Stepwise LR-Scheduler not working across epochs
## Description 

I'm training a model based on number of iterations instead of a number of epochs. The same model trains on datasets of different sizes, hence one epoch differs in the number of iterations. Let's say I want to train e.g. a model for 900 iterations which corresponds to 90 epochs on one of the datasets and want to have a stepwise lr scheduler on iteration 300 & 600. To my understanding this is not natively possible in the pytorch lightning environment.
I know that I can change the lr scheduler interval to "step" and then set the frequency, like so: 

``'lr_scheduler': {"scheduler": sched,
                           "interval": "step",
                           "frequency": 300} ``

However this only applies the steps **within** one epoch. If I set the frequency larger than the number of iteration per epoch no lr step is applied. I would assume that the expected behaviour is to call the scheduler.step() every n frequency step across multiple epochs.

### What version are you seeing the problem on?

v2_0

### How to reproduce the bug

```python
import os

import torch
from torch.utils.data import DataLoader, Dataset
from pytorch_lightning import LightningModule, Trainer


class RandomDataset(Dataset):
    def __init__(self, size, length):
        self.len = length
        self.data = torch.randn(length, size)

    def __getitem__(self, index):
        return self.data[index]

    def __len__(self):
        return self.len


class BoringModel(LightningModule):
    def __init__(self):
        super().__init__()
        self.layer = torch.nn.Linear(32, 2)

    def forward(self, x):
        return self.layer(x)

    def training_step(self, batch, batch_idx):
        loss = self(batch).sum()

        for param_group in self.optimizers().optimizer.param_groups:
            lr = param_group['lr']
        self.log('lr', lr, prog_bar=True, on_step=True, on_epoch=False)
        return {"loss": loss}

    def configure_optimizers(self):
        opt = torch.optim.SGD(self.layer.parameters(), lr=0.1)
        scheduler = torch.optim.lr_scheduler.StepLR(opt, 1)
        return {"optimizer": opt, 'lr_scheduler': {"scheduler": scheduler,
                                                   "interval": "step",
                                                   "frequency": 10}}


def run():
    train_data = DataLoader(RandomDataset(32, 32), batch_size=8)

    model = BoringModel()
    trainer = Trainer(
        accelerator='cpu',
        default_root_dir=os.getcwd(),
        num_sanity_val_steps=0,
        max_epochs=-1,
        max_steps=30,
        log_every_n_steps=1
    )
    trainer.fit(model, train_dataloaders=train_data)


if __name__ == "__main__":
    run()
```
```


### Error messages and logs

_No response_

### Environment

_No response_

### More info

_No response_

### PR DESCRIPTION
**Pull request by falckt**
## What does this PR do?

Fixes #17544

I am not sure whether to consider this as a breaking change. For `frequencies` > `epoch length` I would consider it a bug fix. For `frequencies` < `epoch length` it does change the behavior but IMHO not in a fundamental way and personally I expected the new behavior to begin with.

<details>
  <summary><b>Before submitting</b></summary>

- Was this **discussed/agreed** via a GitHub issue? change according to @awaelchli suggestion in https://github.com/Lightning-AI/pytorch-lightning/issues/17544#issuecomment-1826442106
- [x] Did you read the [contributor guideline](https://github.com/Lightning-AI/lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your **PR does only one thing**, instead of bundling different changes together?
- Did you make sure to **update the documentation** with your changes? 
  
  I don't think it is necessary, I would consider the new behavior the expected one.
- Did you write any **new necessary tests**? 
  
  I couldn't find any tests for the old code. **Q**: does this need a test?
- [x] Did you verify new and **existing tests pass** locally with your changes? it works as expected locally
- Did you list all the **breaking changes** introduced by this pull request? yes
- Did you **update the CHANGELOG**? todo

</details>

## PR review

Anyone in the community is welcome to review the PR.
Before you start reviewing, make sure you have read the [review guidelines](https://github.com/Lightning-AI/lightning/wiki/Review-guidelines). In short, see the following bullet-list:

<details>
  <summary>Reviewer checklist</summary>

- [ ] Is this pull request ready for review? (if not, please submit in draft mode)
- [ ] Check that all items from **Before submitting** are resolved
- [ ] Make sure the title is self-explanatory and the description concisely explains the PR
- [ ] Add labels and milestones (and optionally projects) to the PR so it can be classified

</details>

----
📚 Documentation preview 📚: https://pytorch-lightning--20248.org.readthedocs.build/en/20248/

### CODE REVIEW

**Review by Borda — APPROVED** [review]

**Comment by lantiga** [comment]
Thank you @falckt !
This is a duplicate of #20211 (for fairness the other PR came first : )
maybe you can help with a test there?
